File: java/spring-boot-rabbitmq/rabbitmq-domain/src/test/java/com/meridal/examples/springbootrabbitmq/domain/SongTest.java
package com.meridal.examples.springbootrabbitmq.domain;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SongTest {

    @Test
    public void testSetDurationWithMinsAndSecsMoreThanNine() {
        final Song song = new Song();
        song.setDuration(11, 42);
        assertEquals(song.getDuration(), "11:42");
    }

    @Test
    public void testSetDurationWithMinsLessThanTen() {
        final Song song = new Song();
        song.setDuration(7, 17);
        assertEquals(song.getDuration(), "07:17");
    }

    @Test
    public void testSetDurationWithSecsLessThanTen() {
        final Song song = new Song();
        song.setDuration(23, 2);
        assertEquals(song.getDuration(), "23:02");
    }

    @Test
    public void testSetDurationWithMinsAndSecsLessThanTen() {
        final Song song = new Song();
        song.setDuration(5, 7);
        assertEquals(song.getDuration(), "05:07");
    }

    @Test
    public void testSetDurationWithMinsAndSecsZero() {
        final Song song = new Song();
        song.setDuration(0, 0);
        assertEquals(song.getDuration(), "00:00");
    }
}

File: kotlin/kotlin-domain/src/test/kotlin/com/meridal/examples/domain/SongTest.kt
package com.meridal.examples.domain

import org.junit.Assert.*
import org.junit.Test

class SongTest {

    @Test
    fun testSetDurationWithMinsAndSecsMoreThanNine() {
        val song = Song()
        song.setDuration(11, 42)
        assertEquals("11:42", song.duration)
    }

    @Test
    fun testSetDurationWithMinsLessThanTen() {
        val song = Song()
        song.setDuration(7, 17)
        assertEquals("07:17", song.duration)
    }

    @Test
    fun testSetDurationWithSecsLessThanTen() {
        val song = Song()
        song.setDuration(23, 2)
        assertEquals("23:02", song.duration)
    }

    @Test
    fun testSetDurationWithMinsAndSecsLessThanTen() {
        val song = Song()
        song.setDuration(5, 7)
        assertEquals("05:07", song.duration)
    }

    @Test
    fun testSetDurationWithMinsAndSecsZero() {
        val song = Song()
        song.setDuration(0, 0)
        assertEquals("00:00", song.duration)
    }
}